### PR TITLE
feat: enhance file handling in Input classes

### PR DESCRIPTION
**Pull request by TechyMT**
- Implemented `download_files` method in the `Input` class to streamline file retrieval from local storage or object store.
- Updated `JsonInput` and `ParquetInput` classes to utilize the new `download_files` method, improving file availability checks and error handling.
- Enhanced `get_dataframe` and `get_batched_dataframe` methods to read files directly after ensuring they are downloaded, simplifying the logic and improving maintainability.
- Added detailed docstrings for better clarity on method functionality and expected behavior.

### Changelog

- _to be added_

### Additional context (e.g. screenshots, logs, links)

- _to be added_

### Checklist

- [ ] Additional tests added
- [ ] All CI checks passed
- [ ] Relevant documentation updated

---

### Copyleft License Compliance

- [ ] Have you used any code that is subject to a Copyleft license (e.g., GPL, AGPL, LGPL)?
- [ ] If yes, have you modified the code in the context of this project? please share additional details.

### CODE REVIEW

File: application_sdk/activities/common/utils.py [hunk 1]
```diff
@@ -79,17 +79,62 @@ def build_output_path() -> str:
...
+            # Path is already a relative object store path, normalize and return
+            # Use os.path.normpath to handle cases like ./././path properly
+            normalized = os.path.normpath(path)
+            # If the normalized path is just ".", return empty string for object store
+            result = "" if normalized == "." else normalized
+            # For object store, convert absolute paths to relative by removing leading slash
+            if result.startswith("/"):
+                result = result[1:]
+            # Normalize path separators to forward slashes for object store
+            return result.replace(os.path.sep, "/")
```
[the remark this answers is not in the document]
Comment (TechyMT) [reply application_sdk/activities/common/utils.py:126]: You're correct - UNC paths are not relevant in our Dapr object store context.
The get_object_store_prefix function only handles:
Local file paths → converted to relative object store keys
Object store keys → always relative paths like data/file.txt
Since all object store operations go through Dapr (S3, Azure Blob, etc.), we never encounter UNC network paths like //network/share/file in practice. The current behavior is acceptable because:
UNC paths won't occur in real usage
The security fixes (preventing path traversal) are the critical improvements
If a UNC path were somehow provided, it would still function as a valid object store key
The test case serves more as edge case documentation than a real-world scenario. The important fixes are the os.path.commonpath and os.path.normpath security improvements.

File: application_sdk/activities/common/utils.py [hunk 2]
```diff
@@ -79,17 +79,60 @@ def build_output_path() -> str:
...
+    # Check if path is under TEMPORARY_PATH
+    try:
+        # Use os.path.commonpath to properly check if path is under temp directory
+        # This prevents false positives like '/tmp/local123' matching '/tmp/local'
+        common_path = os.path.commonpath([abs_path, abs_temp_path])
+        if common_path == abs_temp_path:
+            # Path is under temp directory, convert to relative object store path
+            relative_path = os.path.relpath(abs_path, abs_temp_path)
+            # Normalize path separators to forward slashes for object store
+            return relative_path.replace(os.path.sep, "/")
```
Comment (OnkarVO7) [inline application_sdk/activities/common/utils.py:115]: move this into a common method so it can be re-usable

File: application_sdk/activities/metadata_extraction/sql.py [hunk 3]
```diff
@@ -485,7 +485,6 @@ async def transform_data(
 
         raw_input = ParquetInput(
             path=os.path.join(output_path, "raw"),
-            input_prefix=output_prefix,
```
Comment (OnkarVO7) [inline application_sdk/activities/metadata_extraction/sql.py:488]: Breaking change needs to be notified

**Review by OnkarVO7 — COMMENTED** [review]
Minor comments rest looks good

File: application_sdk/activities/common/utils.py (hunk 2 above)
Comment (TechyMT) [reply application_sdk/activities/common/utils.py:152]: done

File: tests/unit/inputs/test_parquet.py [hunk 4]
```diff
@@ -26,382 +23,78 @@ def test_init(config: Dict[str, Any]) -> None:
...
-        mock_glob.return_value = ["existing.parquet"]
-
         parquet_input = ParquetInput(
-            path=path, chunk_size=100000, input_prefix=prefix, file_names=file_names
+            path=path,
+            chunk_size=100000,  # No file_names
         )
 
-        await parquet_input.download_files(path)
+        result = await parquet_input.download_files(".parquet")
```
[the remark this answers is not in the document]
Comment (TechyMT) [reply tests/unit/inputs/test_parquet.py]: done

File: application_sdk/activities/metadata_extraction/sql.py (hunk 3 above)
Comment (TechyMT) [reply application_sdk/activities/metadata_extraction/sql.py:488]: will do them once this pr merged

**Review by inishchith — APPROVED** [review]
